> Patch [1] is wrong. I find it frustrating OpenWRT/LEDE doesn't try to
> work with upstream on ixing these things right.
>---

Thanks, I had to change ar->cal_data to ar->cal_file and add a few
guards around ar->normal_mode_fw.board since it's no longer loaded
and causes the thing to crash otherwise...

With this patch I no longer need the board.bin at all and the device
also initializes.
---
--- a/drivers/net/wireless/ath/ath10k/core.c	2016-11-18 21:58:52.479123559 +0100
+++ b/drivers/net/wireless/ath/ath10k/core.c	2016-11-18 22:15:12.701003098 +0100
@@ -654,6 +654,11 @@ static int ath10k_core_get_board_id_from
 	u8 board_id, chip_id;
 	int ret;
 
+	if (ar->cal_file) {
+		ath10k_dbg(ar, ATH10K_DBG_BOOT, "boot skipping board id, cal data present\n");
+		return 0;
+	}
+
 	address = ar->hw_params.patch_load_addr;
 
 	if (!ar->normal_mode_fw.fw_file.otp_data ||
@@ -1073,6 +1078,11 @@ static int ath10k_core_fetch_board_file(
 	char boardname[100];
 	int ret;
 
+	if (ar->cal_file) {
+		ath10k_dbg(ar, ATH10K_DBG_BOOT, "boot skipping board file, cal data present\n");
+		return 0;
+	}
+
 	ret = ath10k_core_create_board_name(ar, boardname, sizeof(boardname));
 	if (ret) {
 		ath10k_err(ar, "failed to create board name: %d", ret);
--- a/drivers/net/wireless/ath/ath10k/debug.c	2016-10-08 13:44:55.000000000 +0200
+++ b/drivers/net/wireless/ath/ath10k/debug.c	2016-11-18 22:04:57.399832350 +0100
@@ -169,8 +169,8 @@ void ath10k_debug_print_board_info(struc
 	ath10k_info(ar, "board_file api %d bmi_id %s crc32 %08x",
 		    ar->bd_api,
 		    boardinfo,
-		    crc32_le(0, ar->normal_mode_fw.board->data,
-			     ar->normal_mode_fw.board->size));
+		    ar->normal_mode_fw.board ? crc32_le(0, ar->normal_mode_fw.board->data,
+			     ar->normal_mode_fw.board->size) : 0);
 }
 
 void ath10k_debug_print_boot_info(struct ath10k *ar)
@@ -2314,10 +2314,14 @@ static ssize_t ath10k_debug_fw_checksums
 			 "codeswap\t\t%08x\n",
 			 crc32_le(0, ar->normal_mode_fw.fw_file.codeswap_data,
 				  ar->normal_mode_fw.fw_file.codeswap_len));
-	len += scnprintf(buf + len, buf_len - len,
+
+	if (ar->normal_mode_fw.board) {
+		len += scnprintf(buf + len, buf_len - len,
 			 "board-N.bin\t\t%08x\n",
 			 crc32_le(0, ar->normal_mode_fw.board->data,
 				  ar->normal_mode_fw.board->size));
+	}
+
 	len += scnprintf(buf + len, buf_len - len,
 			 "board\t\t\t%08x\n",
 			 crc32_le(0, ar->normal_mode_fw.board_data,
---
